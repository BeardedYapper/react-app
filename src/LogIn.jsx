import React, { useState } from 'react';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button'
import { Link } from 'react-router-dom';
import {useNavigate} from 'react-router-dom';

const axios = require('axios').default;

export default function LogIn ()  {  
  const [email,setEmail] = useState('');
  const [password,setPassword] = useState('');
  const navigate = useNavigate();
  const useLogIn = async() => {
    try {
      const response = await axios.post('http://localhost:3001/authenticate', {
        authenticate: {
          email,
          password
        }
      });
      const data = response.data;
      localStorage.setItem("auth_token", data.auth_token);
      navigate('/')

    } catch (error) {
      console.error(error);
    }
  }
  return (
    <Box
      component="form"
      noValidate
      autoComplete="off"
      style={{marginTop: 'auto'}}
    >
      <div>
        <Typography variant="h3" style={{marginTop: '20px', marginBottom: '20px'}}>
          Log In
        </Typography>
        <div style={{display: 'flex', flexDirection: 'column', alignItems: 'center', margin: '20px'}}>
          <TextField
            required
            id="outlined-required"
            label="Email"
            style={{width:'300px', marginTop: '20px'}}
            value={email}
            onChange={(e)=>setEmail(e.currentTarget.value)}
          />
          <TextField
            required
            id="outlined-required"
            label="Password"
            type="password"
            style={{width:'300px', marginTop: '20px'}}
            value={password}
            onChange={(e)=>setPassword(e.currentTarget.value)}
          />
          <Button onClick={useLogIn} color="success" size="large" style={{ marginTop: '20px'}} variant="contained">Log in</Button>
          <Typography variant="body1" style={{marginTop:'10px'}}>Don't have an account?<Link to="/signUp"> Sign up</Link></Typography>
        </div>
      </div>
    </Box>
  );
}
