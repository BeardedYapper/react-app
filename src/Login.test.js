import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import LogIn from './LogIn';

test('Render LogIn page', () => {
  const { container } = render(
    <Router>
    <LogIn/>
    </Router>,
  );
  expect(container.firstChild).toMatchSnapshot();
});