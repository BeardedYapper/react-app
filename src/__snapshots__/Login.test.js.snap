// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render LogIn page 1`] = `
<form
  autocomplete="off"
  class="MuiBox-root css-0"
  novalidate=""
  style="margin-top: auto;"
>
  <div>
    <h3
      class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
      style="margin-top: 20px; margin-bottom: 20px;"
    >
      Log In
    </h3>
    <div
      style="display: flex; flex-direction: column; align-items: center; margin: 20px;"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        style="width: 300px; margin-top: 20px;"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-required"
          id="outlined-required-label"
        >
          Email
          <span
            aria-hidden="true"
            class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-required"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Email
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        style="width: 300px; margin-top: 20px;"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-required"
          id="outlined-required-label"
        >
          Password
          <span
            aria-hidden="true"
            class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-required"
            required=""
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Password
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-mkazzp-MuiButtonBase-root-MuiButton-root"
        style="margin-top: 20px;"
        tabindex="0"
        type="button"
      >
        Log in
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        style="margin-top: 10px;"
      >
        Don't have an account?
        <a
          href="/signUp"
        >
           Sign up
        </a>
      </p>
    </div>
  </div>
</form>
`;
